### DISCUSSION TITLE
How to get callback when user taps on the payment sheet button

### DISCUSSION THREAD

**Discussion by appeventic** [flutter-stripe/flutter_stripe#1557] [Q&A]
**Describe the bug**
In `customFlow:flase` I need to save a payment transaction when user taps on the payment sheet button, is there a way to get a callback? I want to submit the payment transaction before I submit it to Stripe. I want to get the callback for both platform pay or card payment options.

<img width="351" alt="Screenshot 2023-12-15 at 11 57 34 AM" src="https://github.com/flutter-stripe/flutter_stripe/assets/141521094/7534b707-d611-4a03-843b-a97339d4cbc5">

**Comment by remonh87** [comment]
This is not possible for security reasons. What is the use case for needing a pre pay callback?

**Reply by appeventic** [reply]
I need to save a payment transaction object in the database, so need the callback to make sure user actually making the payment. User can always cancel the bottom sheet, so without the callback there is no way to know whether user finally paying or not.
